R/pkg/R/SQLContext.R

```diff
@@ -17,6 +17,34 @@
 
 # SQLcontext.R: SQLContext-driven functions
 
+#' Temporary function to reroute old S3 Method call to new
+#' We need to check the class of x to ensure it is SQLContext before dispatching
+dispatchFunc <- function(newFuncSig, x, ...) {
+  funcName <- as.character(sys.call(sys.parent())[[1]])
+  f <- get0(paste0(funcName, ".default"))
+  # Strip sqlContext from list of parameters and then pass the rest along.
+  # In the following, if '&' is used instead of '&&', it warns about
+  # "the condition has length > 1 and only the first element will be used"
+  if (class(x) == "jobj" &&
+      grepl("org.apache.spark.sql.SQLContext", capture.output(show(x)))) {
+    .Deprecated(newFuncSig, old = paste0(funcName, "(sqlContext...)"))
+    f(...)
+  } else {
+    f(x, ...)
+  }
+}
```

Comment:
The proposal for this is to eliminate the sqlContext parameter from SQLContext-parity methods in R. Primarily this makes methods friendlier in R and more R-like (eg. read.df()). The changed method signature would be the one we would like to keep in the next release.

Reasons for this have been discussed in JIRA, but to recap:
1. We only support one sqlContext in R - and having multiple at a time can be very confusing (eg. table not accessible)
2. For hiveCtx vs sqlContext, hiveCtx is preferred